### PR TITLE
[5.1] No of retries left functionality added to ThrottlesLogins

### PR DESCRIPTION
**Pull request by knvpk**
It is helpfull have like this function. see [Issue](https://github.com/laravel/framework/issues/10326).

### CODE REVIEW

File: src/Illuminate/Cache/RateLimiter.php [hunk 1]
```diff
@@ -52,6 +52,17 @@ public function tooManyAttempts($key, $maxAttempts, $decayMinutes = 1)
     }
 
     /**
+     * Get the number of attempts using key.
+     *
+     * @param $key
```
Comment (GrahamCampbell) [inline src/Illuminate/Cache/RateLimiter.php]: cs

File: src/Illuminate/Foundation/Auth/ThrottlesLogins.php [hunk 2]
```diff
@@ -23,6 +23,17 @@ protected function hasTooManyLoginAttempts(Request $request)
     }
 
     /**
+     * Determine how many retries left.
+     *
+     * @param Request $request
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Auth/ThrottlesLogins.php]: cs

File: src/Illuminate/Foundation/Auth/ThrottlesLogins.php [hunk 3]
```diff
@@ -23,6 +23,17 @@ protected function hasTooManyLoginAttempts(Request $request)
...
+     * @return int
+     */
+    protected function retriesLeft(Request $request)
+    {
+        return $this->maxLoginAttempts() - app(RateLimiter::class)->attempts($request->input($this->loginUsername()).$request->ip()) + 1;
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Auth/ThrottlesLogins.php]: Could you make this line shorted by assigning variables earlier within this method please.

**Comment by pavankumarkatakam** [comment]
> Could you make this line shorted by assigning variables earlier within this method please.

I didn't understand.

**Comment by JosephSilber** [comment]
@pavankumarkatakam Likie so:

```php
$key = $request->input($this->loginUsername()).$request->ip();

$attempts = app(RateLimiter::class)->attempts($key);

return $this->maxLoginAttempts() - $attempts + 1;
```

**Comment by GrahamCampbell** [comment]
StyleCI is so useful. :)

**Pull request merged**